### DISCUSSION TITLE
django.db.utils.ProgrammingError: column django_q_schedule.intended_date_kwarg does not exist

### DISCUSSION THREAD

**Discussion by swaan** [inventree/InvenTree#6168] [Q&A] [answered]
After updating containers to the latest version I get this error 
[_inventree-production-inventree-server-1_logs.txt](https://github.com/inventree/InvenTree/files/13854517/_inventree-production-inventree-server-1_logs.txt)
postgres logs: 
[_inventree-production-inventree-db-1_logs.txt](https://github.com/inventree/InvenTree/files/13854590/_inventree-production-inventree-db-1_logs.txt)

I'm running Armbian 23.11.1 with 5.10.160 kernel. Docker version 24.0.7, build afdd53b. I'm using the production compose file with only the backup dir changed.
I'd appreciate any pointers.

**Comment by matmair** [comment, marked as answer]
Looks like there was a problem with the update script step (https://docs.inventree.org/en/stable/start/docker_prod/#update-database) - please repeat all the steps int the [updating guide for docker](https://docs.inventree.org/en/stable/start/docker_prod/#updating-inventree).

**Reply by swaan** [reply]
You were correct. I didn't realize but the backup function had failed (path not mounted) and thought backup was an optional step. Now that I ran update with --skip-backup the process didn't stop there and now everything is working again. Thanks!

**Reply by swaan** [reply]
Maybe this is a very rare situation but it would be helpful to print "Aborting" when the process requires backup to complete and it doesn't.

**Reply by matmair** [reply]
There should actually be a warning about the path not exsisting - can you check the logs again and open a bug if it is missing?

**Reply by swaan** [reply]
This is the update script with the attempted backup after the last line it exited:

> Backing up InvenTree database...
> Python version 3.10.13 - /usr/local/bin/python3
> Backing Up Database: inventree
> Writing file to default-ccdbccb74314-2024-01-08-214249.psql.bin.gz
> PermissionError: [Errno 1] Operation not permitted: '/home/inventree/data/ja/default-ccdbccb74314-2024-01-08-214249.psql.bin.gz'
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/utils.py", line 120, in wrapper
>     func(*args, **kwargs)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/management/commands/dbbackup.py", line 93, in handle
>     self._save_new_backup(database)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/management/commands/dbbackup.py", line 120, in _save_new_backup
>     self.write_to_storage(outputfile, filename)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/management/commands/_base.py", line 97, in write_to_storage
>     self.storage.write_file(file, path)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/storage.py", line 86, in write_file
>     self.storage.save(name=filename, content=filehandle)
>   File "/root/.local/lib/python3.10/site-packages/django/core/files/storage.py", line 54, in save
>     name = self._save(name, content)
>   File "/root/.local/lib/python3.10/site-packages/django/core/files/storage.py", line 279, in _save
>     fd = os.open(full_path, self.OS_OPEN_FLAGS, 0o666)
> 
> Traceback (most recent call last):
>   File "/home/inventree/InvenTree/manage.py", line 23, in <module>
>     execute_from_command_line(sys.argv)
>   File "/root/.local/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
>     utility.execute()
>   File "/root/.local/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
>     self.fetch_command(subcommand).run_from_argv(self.argv)
>   File "/root/.local/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
>     self.execute(*args, **cmd_options)
>   File "/root/.local/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
>     output = self.handle(*args, **options)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/utils.py", line 120, in wrapper
>     func(*args, **kwargs)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/management/commands/dbbackup.py", line 93, in handle
>     self._save_new_backup(database)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/management/commands/dbbackup.py", line 120, in _save_new_backup
>     self.write_to_storage(outputfile, filename)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/management/commands/_base.py", line 97, in write_to_storage
>     self.storage.write_file(file, path)
>   File "/root/.local/lib/python3.10/site-packages/dbbackup/storage.py", line 86, in write_file
>     self.storage.save(name=filename, content=filehandle)
>   File "/root/.local/lib/python3.10/site-packages/django/core/files/storage.py", line 54, in save
>     name = self._save(name, content)
>   File "/root/.local/lib/python3.10/site-packages/django/core/files/storage.py", line 279, in _save
>     fd = os.open(full_path, self.OS_OPEN_FLAGS, 0o666)
> PermissionError: [Errno 1] Operation not permitted: '/home/inventree/data/ja/default-ccdbccb74314-2024-01-08-214249.psql.bin.gz'

/home/inventree/data/ja/ has the immutable flag set as it should not write there when its not mounted.

Edit: with the traceback it should have been obvious to me that the failed backup was the reason the process ended.
